From wfp5p@virginia.edu  Mon Jul 27 14:07:41 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 27 Jul 2009 16:47:28 -0400
Subject: Staging: hv: fix blkvsc_release() parameters
To: greg@kroah.com
Cc: hjanssen@microsoft.com
Message-ID: <1248727665-23969-7-git-send-email-wfp5p@virginia.edu>


blkvsc_release() had the wrong parameter list for  struct
block_device_operations

Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Cc: Hank Janssen <hjanssen@microsoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/hv/blkvsc_drv.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/drivers/staging/hv/blkvsc_drv.c
+++ b/drivers/staging/hv/blkvsc_drv.c
@@ -125,7 +125,7 @@ static int blkvsc_remove(struct device *
 static void blkvsc_shutdown(struct device *device);
 
 static int blkvsc_open(struct block_device *bdev,  fmode_t mode);
-static int blkvsc_release(struct inode *inode, struct file *filep);
+static int blkvsc_release(struct gendisk *disk, fmode_t mode);
 static int blkvsc_media_changed(struct gendisk *gd);
 static int blkvsc_revalidate_disk(struct gendisk *gd);
 static int blkvsc_getgeo(struct block_device *bd, struct hd_geometry *hg);
@@ -1341,9 +1341,9 @@ static int blkvsc_open(struct block_devi
 	return 0;
 }
 
-static int blkvsc_release(struct inode *inode, struct file *filep)
+static int blkvsc_release(struct gendisk *disk, fmode_t mode)
 {
-	struct block_device_context *blkdev = inode->i_bdev->bd_disk->private_data;
+	struct block_device_context *blkdev = disk->private_data;
 
 	DPRINT_DBG(BLKVSC_DRV, "- users %d disk %s\n", blkdev->users, blkdev->gd->disk_name);
 
